db/migrate/20161202161114_add_files_collection_ref_to_tribunal_case.rb

```diff
@@ -0,0 +1,5 @@
+class AddFilesCollectionRefToTribunalCase < ActiveRecord::Migration[5.0]
+  def change
+    add_column :tribunal_cases, :files_collection_ref, :uuid, default: 'uuid_generate_v4()', unique: true
```

Comment:
Yes, that was the main reason. I think if we were to know in advance the user will always have to complete a step where only 1 file is uploaded, and said file is mandatory in order to continue, then we could happily let the uploader generate the collection uuid and return it back to store in the tribunal case table, and following steps involving file uploads will use that previously generated and stored uuid.

But I think that is a big assumption. At the moment we have a step that allows to upload 1 file and only 1, but it is not mandatory to upload a file at said step, so the user could arrive to the multiple (parallel when using javascript dropzone) file upload without a generated collection uuid.

But I agree this might be prone to be improved at some point, once we have advance more in the data capture steps.